apps/rush-lib/src/utilities/Utilities.ts

```diff
@@ -548,23 +622,42 @@ export class Utilities {
         continue;
       }
 
-      environment[key] = initialEnvironment[key];
+      environment[key] = options.initialEnvironment[key];
     }
 
-     // When NPM invokes a lifecycle script, it sets an environment variable INIT_CWD that remembers
-     // the directory that NPM started in.  This allows naive scripts to change their current working directory
-     // and invoke NPM operations, while still be able to find a local .npmrc file.  Although Rush recommends
-     // for toolchain scripts to be professionally written (versus brittle stuff like
-     // "cd ./lib && npm run tsc && cd .."), we support INIT_CWD for compatibility.
-     //
-     // More about this feature: https://github.com/npm/npm/pull/12356
-     if (initCwd) {
-       environment['INIT_CWD'] = initCwd; // tslint:disable-line:no-string-literal
-     }
+    // When NPM invokes a lifecycle script, it sets an environment variable INIT_CWD that remembers
+    // the directory that NPM started in.  This allows naive scripts to change their current working directory
+    // and invoke NPM operations, while still be able to find a local .npmrc file.  Although Rush recommends
+    // for toolchain scripts to be professionally written (versus brittle stuff like
+    // "cd ./lib && npm run tsc && cd .."), we support INIT_CWD for compatibility.
+    //
+    // More about this feature: https://github.com/npm/npm/pull/12356
+    if (options.initCwd) {
+      environment['INIT_CWD'] = options.initCwd; // tslint:disable-line:no-string-literal
+    }
+
+    if (options.pathOptions) {
+      if (options.pathOptions.includeRepoBin && options.pathOptions.commonTempFolder) {
+        environment.PATH = Utilities.addNodeModulesBinToPATH(environment.PATH, options.pathOptions.commonTempFolder);
+      }
+
+      if (options.pathOptions.includeProjectBin && options.pathOptions.projectRoot) {
+        environment.PATH = Utilities.addNodeModulesBinToPATH(environment.PATH, options.pathOptions.projectRoot);
+      }
+    }
 
     return environment;
   }
 
+  private static addNodeModulesBinToPATH(existingPATH: string | undefined, rootDirectory: string): string {
+    const binPath: string = path.resolve(rootDirectory, 'node_modules', '.bin');
+    if (existingPATH) {
+      return `${binPath}:${existingPATH}`;
```

Comment:
Windows uses `;` instead of `:` as the `PATH` separator.  I feel like we handle this elsewhere in **rush-lib**.

Reply:
Yeah, see [BaseRushAction.ts](https://github.com/Microsoft/web-build-tools/blob/577efd3d2251e72649aa8f7b7a475ea7b77132a2/apps/rush-lib/src/cli/actions/BaseRushAction.ts#L91) and [Executable.ts](https://github.com/Microsoft/web-build-tools/blob/577efd3d2251e72649aa8f7b7a475ea7b77132a2/libraries/node-core-library/src/Executable.ts#L358).

Reply:
I'm not seeing anywhere, but I'll change it.

Reply:
The two things I liked are using `path.delimiter`

Reply:
Ah didn't see that. I'll change it.